### ISSUE TITLE
Update to latest SDK

### ISSUE THREAD

**Issue by galvesribeiro** [alfonsosanchezbeato/jetson-ubuntu-core#4]
Hello! Thank you very much for providing this port!

I've followed your posts and was able to build the UC image just fine even thought it is only working on a real Ubuntu 18 machine/VM with Snapcraft on it, and with the latest snapcraft it requires us to use `--destructive-mode` otherwise it would never cross-compile it.

The SDK being used is very old and I was trying to change it to make it work on the current SDK `32.4.3`. However, I'm failing to get it to work because the `source_sync.sh` scripts takes 1 single tag for all the nvidia repos it will clone.

An example of this is the BCT repo:

```
Downloading default hardware/nvidia/platform/t19x/galen/bct source...
Cloning into '/home/guto/uc-test/jetson-kernel-snap/nano/parts/kernel/src/sources/hardware/nvidia/platform/t19x/galen/bct'...
remote: Enumerating objects: 389, done.
remote: Counting objects: 100% (389/389), done.
remote: Compressing objects: 100% (237/237), done.
remote: Total 389 (delta 271), reused 224 (delta 150)
Receiving objects: 100% (389/389), 259.74 KiB | 256.00 KiB/s, done.
Resolving deltas: 100% (271/271), done.
The default hardware/nvidia/platform/t19x/galen/bct source is downloaded in: /home/guto/uc-test/jetson-kernel-snap/nano/parts/kernel/src/sources/hardware/nvidia/platform/t19x/galen/bct
Couldn't find tag tegra-l4t-r32.4.3
/home/guto/uc-test/jetson-kernel-snap/nano/parts/kernel/src/sources/hardware/nvidia/platform/t19x/galen/bct source sync to tag tegra-l4t-r32.4.3 failed!
```

All the other repos that are part of the kernel snap are synced and have that `32.4.3` tag on them but the [BCT doesn't](https://nv-tegra.nvidia.com/gitweb/?p=device/hardware/nvidia/platform/t19x/stardust-bct.git;a=summary).

Also, even using your repo without any changes, it looks like the NVidia libraries are not installed on the final image making it unusable to run things that leverage CUDA or the NVidia Container Runtime on it.

I was about to submit a PR against each repo updating them but I got stuck on it.

Do you have any idea what can be done to get it to work?

Thanks!

**Comment by alfonsosanchezbeato** [comment]
Yes, the nvidia libraries are not included in the snaps that compose the Ubuntu Core image. This is expected, as in a snap-only system usually each snap carries its own dependencies. You can take a look at https://github.com/alfonsosanchezbeato/jetson-nano-x11 to see an example of how this works: the snap itself will contain all needed libs, including CUDA, for instance.

I am more than happy to review PRs to update to a newer SDK version :-)

**Comment by galvesribeiro** [comment]
> Yes, the nvidia libraries are not included in the snaps that compose the Ubuntu Core image. This is expected, as in a snap-only system usually each snap carries its own dependencies. You can take a look at https://github.com/alfonsosanchezbeato/jetson-nano-x11 to see an example of how this works: the snap itself will contain all needed libs, including CUDA, for instance.

Ok will look at it.

> I am more than happy to review PRs to update to a newer SDK version :-)

I wish I could add it. But like I said BCT doesn't have the tag for the latest kernel 😦
